modules/account/src/main/kotlin/edu/stanford/spezi/module/account/cred/manager/CredentialManagerAuth.kt

```diff
@@ -0,0 +1,143 @@
+import com.google.android.libraries.identity.googleid.GoogleIdTokenParsingException
+import edu.stanford.spezi.core.logging.speziLogger
+import edu.stanford.spezi.core.navigation.DefaultNavigationEvent
+import edu.stanford.spezi.core.navigation.Navigator
+import edu.stanford.spezi.module.account.AccountNavigationEvent
+import edu.stanford.spezi.module.account.R
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+class CredentialManagerAuth @Inject constructor(
+    private val credentialManager: CredentialManager,
+    private val firebaseAuthManager: FirebaseAuthManager,
+    private val navigator: Navigator,
+) : SignInHandler {
+    private val logger by speziLogger()
+
+    override fun handleSignIn(
+        context: Context,
+        scope: CoroutineScope,
+        updateUiState: (GoogleIdTokenCredential) -> Unit,
+    ) {
+        scope.launch {
+            var result = getCredential(context, true)
+            if (result == null) {
+                logger.i { "No authorized accounts found" }
+                result = getCredential(context, false)
+            }
+            result?.let { handleSignIn(it, updateUiState) }
+        }
+    }
+
+    private suspend fun getCredential(
+        context: Context,
+        filterByAuthorizedAccounts: Boolean,
+    ): GetCredentialResponse? {
+        val googleIdOption = GetGoogleIdOption.Builder()
+            .setFilterByAuthorizedAccounts(filterByAuthorizedAccounts)
+            .setServerClientId(context.getString(R.string.serverClientId))
+            .build()
+
+        val request = GetCredentialRequest.Builder()
+            .addCredentialOption(googleIdOption)
+            .build()
+
+        return try {
+            credentialManager.getCredential(
+                request = request,
+                context = context
+            )
+        } catch (e: GetCredentialException) {
+            logger.e { "Error getting credential: ${e.message}" }
+            null
+        }
+    }
+
+    private suspend fun handleSignIn(
+        result: GetCredentialResponse,
+        updateUiState: (GoogleIdTokenCredential) -> Unit,
+    ) {
+        when (val credential = result.credential) {
+            is CustomCredential -> {
+                if (credential.type == GoogleIdTokenCredential.TYPE_GOOGLE_ID_TOKEN_CREDENTIAL) {
+                    try {
+                        val googleIdTokenCredential =
+                            GoogleIdTokenCredential.createFrom(credential.data)
+                        logger.i { "Display Name: ${googleIdTokenCredential.displayName}" }
+                        logger.i { "Phone Number: ${googleIdTokenCredential.phoneNumber}" }
+                        logger.i { "Family Name: ${googleIdTokenCredential.familyName}" }
+                        updateUiState(googleIdTokenCredential)
+
+                        if (isNewUser(googleIdTokenCredential)) {
```

Comment:
Let's sync tomorrow how to simplify these checks, currently `firebaseAuthManager.checkIfNewUser` is also attempting to sing in which is what we actually want to, but due to the function name `isNewUser` that looks like a side effect instead